M2/Macaulay2/m2/run.m2

```diff
@@ -128,29 +142,25 @@ runFile = (inf, inputhash, outf, tmpf, pkg, announcechange, usermode, examplefil
      cmd = cmd | readmode(SetInputFile,   "<" | format inf);
      cmd = cmd | readmode(SetOutputFile,  ">>" | format toAbsolutePath tmpf);
      cmd = cmd | readmode(SetCaptureErr,  "2>&1");
-     if debugLevel > 0 then stderr << endl << cmd << endl;
      for fn in examplefiles do copyFile(fn,rundir | baseFilename fn);
      r := run cmd;
      if r == 0 then (
 	  scan(reverse findFiles rundir, f -> if isDirectory f then removeDirectory f else removeFile f);
 	  moveFile(tmpf,outf);
 	  return true;
 	  );
-     if debugLevel == 0 then stderr << endl;
-     stderr << cmd << endl;
-     stderr << (new FilePosition from (tmpf, 0, 1)) << ":(" << loadDepth << "):[" << recursionDepth() << "]: (output file) error: Macaulay2 " << describeReturnCode r << endl;
-     stderr << aftermatch(M2errorRegexp,get tmpf);
-     infcontents := get inf;
-     stderr << (new FilePosition from (inf, 0, 1)) << ": (input file)" << endl;
-     scan(statusLines infcontents, x -> stderr << x << endl);
-     m := regex("^--([^\n]+): location of test code", infcontents);
-     if m =!= null then stderr << (
-	 substring(m#1, infcontents)) << ": (location of test code)" << endl;
-     if # findFiles rundir == 1
-     then removeDirectory rundir
+     -- print debugging info
+     printerr "running failed, too!";
+     printerr("error log:  " | logLocation tmpf);
+     printerr("test input: " | testLocation inf);
```

Comment:
Isn't `runFile` used for both tests and examples?  Is this too test-specific?

Perhaps there could be a test-specific method that calls `runFile`, and if there was an error, print this debugging info.  And if that method already knows the `TestInput` object, we wouldn't need to worry about also storing the file location in the test string and using a regex to extract it.

Reply:
That sounds great, but I'll be honest, I hate `runFile` and I don't want to be the one to separate it out XD I wrote this PR because I was simply _too_ frustrated that C-c C-c didn't work and that the test location wasn't printed anymore.

But I will happily change "test input" to something else in a new commit and skip CI.

Reply:
Fair enough :)  Maybe I'll tackle `runFile` this summer.